src/main/java/pl/edu/agh/mwo/Scanner.java

```diff
@@ -3,6 +3,10 @@
 public class Scanner {
 
 	String text;
+	static final int substrinBeginIndex = 0;
+	static final int substringEndIndex = 1;
+	static final int substringNewStringIndex = 1;
```

Comment:
Te stałe nie są raczej potrzebne. Ich wartości wynikają niejako z idei algorytmu. Przykład: jeśli mamy sytuację typu firstChar = text.substring(0, 1) to przecież nie będziemy zmieniać tych stałych.